### DISCUSSION TITLE
I got error when I tried fetch or axios in api handler in App Router.

### DISCUSSION THREAD

**Discussion by melodyxpot** [vercel/next.js#54569] [App Router]
Hello everyone.
As a Next.JS fan, I was building the project using that but I had some problem.
When I tried to use axios or fetch method to pull the data from other api
![image](https://github.com/vercel/next.js/assets/53822027/d6d5f287-e602-4140-9278-224549e72ea1)
This is the error log I had.
![image](https://github.com/vercel/next.js/assets/53822027/17a458c1-f170-41ad-96b6-ff6978066a47)
Please check and give me the solution.
Cheers

**Comment by devjiwonchoi** [comment]
Not sure why `NextResponse` is not returned inside `then`, but I do know the reason for the error.
It is because your handler is not returning any Response.

Try this code:

```ts
import type { NextApiHandler } from 'next'
import { NextResponse } from 'next/server'
import axios from 'axios'

export const GET: NextApiHandler = async (req, res) => {
  try {
    const response = await axios.get(
      'https://jsonplaceholder.typicode.com/posts'
    )
    return NextResponse.json({ data: response.data }, { status: 200 })
  } catch (error) {
    if (axios.isAxiosError(error)) {
      return NextResponse.json({ error: error.message }, { status: 500 })
    } else {
      throw error
    }
  }
}
```

I'll investigate why it is not returned inside `then` and let you know if I figure it out.

P.S. I'm surprised we could use NextApiHandler in `route.ts` 🤯.

**Reply by devjiwonchoi** [reply]
> Not sure why NextResponse is not returned inside `then`

Credits to `unstable_alfon ()` in Discord:

> its basically because the axios.get function is not awaited. and API handler does not run indefinetly/it might be that its not persisted in the memory
axios is run syncrhonously so it just went straight pass through and since there isn't any NextResponse return in GET function, it will throw 500 Internal server error

**Comment by icyJoseph** [comment]
Your GET function is returning undefined.

This is somewhat of an equivalent.

```js
export const GET = async () => {
   axios.get(/* rest of axios stuff */)

   return undefined
}
```

You can rewrite it to:

```js
export const GET = async () => axios.get(/* rest of your stuff */)
```

**Reply by devjiwonchoi** [reply]
😱 Wow, what a shame I missed that. Thank you!!
